tests/testCommandBasedBackendUnified.cpp

```diff
@@ -256,6 +256,46 @@ struct ArrayFloatSingleLine : public RegisterDescriptorBase {
 
 /**********************************************************************************************************************/
 
+struct ArrayHexMultiLine : public RegisterDescriptorBase {
+  static std::string path() { return "/myHex"; }
+  static bool isWriteable() { return true; }
+  static size_t nElementsPerChannel() { return 3; }
+
+  using minimumUserType = uint64_t;
+
+  template<typename UserType>
+  std::vector<std::vector<UserType>> generateValue([[maybe_unused]] bool raw = false) {
+    assert(!raw);
+    std::vector<UserType> v(nElementsPerChannel());
+    for(uint32_t e = 0; e < nElementsPerChannel(); ++e) {
+      v[e] = userTypeToUserType<UserType>(dummyServer.hex[e]) + 17 + 3 * e;
+    }
+    return {v};
+  }
+
+  template<typename UserType>
+  std::vector<std::vector<UserType>> getRemoteValue([[maybe_unused]] bool raw = false) {
+    assert(!raw);
+    std::vector<UserType> v(nElementsPerChannel());
+    for(uint32_t e = 0; e < nElementsPerChannel(); ++e) {
+      v[e] = userTypeToUserType<UserType>(dummyServer.hex[e]);
```

Comment:
Aha, that's why the test passes. It does not require the text to be a hex representation. This is just taking the integer string, interprets it as integer and returns the correct number.
 It is missing force0xHex() here.
 
 I am wondering if we should interpret the string and store the value, and then getRemoteValue() returns the stored value. In principle it does not matter where we interpret the string. However, we also need it in setRemoteValue, so we should do it in  DummyServer::setHex().